common/services/move.js

```diff
@@ -269,8 +269,7 @@ const moveService = {
       .update('move', {
         id,
         status: 'cancelled',
-        cancellation_reason: reason,
```

Comment:
I think it would be better to keep the structure of this method the same so that the names of the API attributes are kept in one place and isolated.

By changing this we have made the controllers need to know about the API attribute names and if this ever changed we'd be having to update it in a number of different places.

The idea of these services was to separate that logic so that only the service needs to care about the API payload structures.

Reply:
However, I know the _create_ does slightly contradict this concept...but I think it would be nice to not change this if possible.

Reply:
So the two cancel methods (moves, allocations) were previously doing this differently. 

Allocation was passing the data all the way through, whereas move was "renaming" the properties.

https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blame/5093d71a7a2a4e3d5de5ed90257979eabdd0835d/app/allocation/controllers/cancel.js

https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blame/5093d71a7a2a4e3d5de5ed90257979eabdd0835d/app/move/controllers/cancel.js

This change was aiming to simplify this logic around conditional parameters, but it was made at the expense of bleeding through architectural tiers.

Based upon your comments above, how about I make a separate PR to change allocations to have a specific data structure - that should simplify the changes needed in this one.

Reply:
>Based upon your comments above, how about I make a separate PR to change allocations to have a specific data structure - that should simplify the changes needed in this one.

This sounds like a good plan to me. I do understand in some areas we've bleed the architecture of the API into the form wizard and controllers but I think it would be nice to try to continue to avoid that where possible.

Reply:
The refactoring PR is https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/pull/802

Reply:
Refactoring PR has been merged, so controllers now don't pass through properties into the service.